src/main/java/io/jenkins/plugins/artifact_manager_jclouds/JCloudsArtifactManager.java

```diff
@@ -123,31 +125,62 @@ public void archive(FilePath workspace, Launcher launcher, BuildListener listene
             String blobPath = getBlobPath(path);
             Blob blob = blobStore.blobBuilder(blobPath).build();
             blob.getMetadata().setContainer(provider.getContainer());
+            String contentType = workspace.act(new ContentTypeGuesser(entry.getKey()));
+            if (contentType != null) {
+                contentTypes.put(entry.getValue(), contentType);
+            }
+            blob.getMetadata().getContentMetadata().setContentType(contentType);
             artifactUrls.put(entry.getValue(), provider.toExternalURL(blob, HttpMethod.PUT));
         }
 
-        workspace.act(new UploadToBlobStorage(artifactUrls, listener));
+        workspace.act(new UploadToBlobStorage(artifactUrls, contentTypes, listener));
         listener.getLogger().printf("Uploaded %s artifact(s) to %s%n", artifactUrls.size(), provider.toURI(provider.getContainer(), getBlobPath("artifacts/")));
     }
 
+    private static class ContentTypeGuesser extends MasterToSlaveFileCallable<String> {
+        private static final long serialVersionUID = 5743824603401251506L;
+
+        private final String relPath;
+
+        ContentTypeGuesser(String relPath) {
+            this.relPath = relPath;
+        }
+
+        @Override
+        public String invoke(File f, VirtualChannel channel) {
+            try {
+                File theFile = new File(f, relPath);
+                String contentType = Files.probeContentType(theFile.toPath());
+                if (null == contentType) {
+                    contentType = URLConnection.guessContentTypeFromName(theFile.getName());
+                }
+                return contentType;
+            } catch (IOException e) {
+                return null;
+            }
+        }
+    }
+
     private static class UploadToBlobStorage extends MasterToSlaveFileCallable<Void> {
         private static final long serialVersionUID = 1L;
 
         private final Map<String, URL> artifactUrls; // e.g. "target/x.war", "http://..."
+        private final Map<String, String> contentTypes;
```

Comment:
Perhaps add a comment example here like the field above, showing what this is mapping from/to?

Reply:
Will do.